### PR TITLE
[#226] Update to manage secrets with Github Environments for Staging and Production

### LINKED ISSUE #226: Update to manage secrets with Github Environments for Staging and Production
## Why

Currently, both the `root template` and `template project` manage to define all variables for GitHub workflows for both Production and Staging env in the same place [Secrets > Action](https://github.com/nimblehq/flutter-templates/settings/secrets/actions). Hence, we must manually name these variables with `_PRODUCTION` or `_STAGING` postfixes and can not really know which one belongs to each env or is being shared between them.

image

GitHub supports a feature to manage them in each environment called [`Environments`](https://github.com/nimblehq/flutter-templates/settings/environments). We should follow this approach to define them wisely.
- Variables belonging to the app build will be moved to [`Environments`](https://github.com/nimblehq/flutter-templates/settings/environments).
- Variables only be used for the workflow in both env will be kept in [Secrets > Action](https://github.com/nimblehq/flutter-templates/settings/secrets/actions).
 
## Who Benefits?

Developers

### PR DESCRIPTION
**Pull request by luongvo**
- Close #226

## What happened 👀

- Remove unnecessary token when checking out in bump_version workflow.
- Fix: exporting multi-lines env files using quote "".
- Convert some secrets to variables.
- Move some secrets to the `Environments` config.

## Insight 📝

- To [echo a multi-line variable](https://github.com/nimblehq/flutter-templates/pull/254#discussion_r1280079272) in GitHub Actions, use quote `""` https://stackoverflow.com/a/22101842/2100084 ✅ 
- `ENV_PRODUCTION` and `ENV_STAGING` have been moved to the `Environments` ✅ 
- `FIREBASE_ANDROID_APP_ID_PRODUCTION`, `FIREBASE_ANDROID_APP_ID_STAGING`, `FIREBASE_IOS_APP_ID_PRODUCTION`, and `FIREBASE_IOS_APP_ID_STAGING` have been converted to variables and moved to the `Environments` ✅ 
- `ANDROID_SIGNING_PROPERTIES_BASE64` is now become `ANDROID_SIGNING_PROPERTIES` without base64 encoded by using `a multi-line variable` ✅ 

## Proof Of Work 📹

- The `bump_version` workflow works properly without token to check out https://github.com/nimblehq/flutter-templates/actions/runs/5746348294/job/15575671352 👉  https://github.com/nimblehq/flutter-templates/pull/264 ✅ 
- Updated secrets & variables management by using `Environment`

  

  

- The app can read secrets properly from .env

### CODE REVIEW

File: bricks/template/__brick__/{{project_name.snakeCase()}}/ios/fastlane/Constants/Environments.rb
Comment (doannimble) [inline bricks/template/__brick__/{{project_name.snakeCase()}}/ios/fastlane/Constants/Environments.rb:1]: @luongvo Could you please add this one into the Github Action secret too?

```
def self.APPSTORE_CONNECT_API_KEY
    ENV['APPSTORE_CONNECT_API_KEY']
  end
```

Comment (luongvo) [reply bricks/template/__brick__/{{project_name.snakeCase()}}/ios/fastlane/Constants/Environments.rb:1]: @doannimble added from `1Pass > iOS > iOS App Delivery Variables`
image

**Review by doannimble — APPROVED** [review]
We may need to check and update writing multiple lines later. Look good so far 👍

**Comment by doannimble** [comment]
> > We may need to check and update writing multiple lines later. Look good so far 👍
> 
> @doannimble It has been verified to work properly with this .env structure https://github.com/nimblehq/flutter-templates/blob/develop/sample/.env.sample ✅
> 
> image

Nice @luongvo . Actually, it can write and read, but it may be incorrect. Have you tried to print out `SECRET` and `REST_API_ENDPOINT` (on app) to check (with the build made by github action)?.

**Comment by luongvo** [comment]
@doannimble Yes, the last build can not run to show the staging name with a black screen https://github.com/nimblehq/flutter-templates/blob/b2851fdae0687015c2fb1ca95f2d8b3205df1a17/sample/lib/main.dart#L117-L123

image

**Review by nkhanh44 — APPROVED** [review]

**Pull request merged**